group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Record.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group;
+
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+
+import java.util.Objects;
+
+/**
+ * A Record which contains an {{@link ApiMessageAndVersion}} as key and
+ * an {{@link ApiMessageAndVersion}} as value. The value could be null.
+ *
+ * This class is immutable.
+ */
+public class Record {
+    /**
+     * The key of the record.
+     */
+    private final ApiMessageAndVersion key;
+
+    /**
+     * The value of the record or null if the record is
+     * a tombstone.
+     */
+    private final ApiMessageAndVersion value;
+
+    /**
+     * Constructs a Record.
+     *
+     * @param key   A non-null key.
+     * @param value A key or null.
+     */
+    public Record(
+        ApiMessageAndVersion key,
+        ApiMessageAndVersion value
+    ) {
+        this.key = Objects.requireNonNull(key);
+        this.value = value;
+    }
+
+    /**
+     * Returns the key.
```

Comment:
nit: Do we need both of these?

Reply:
I never know what to put for getters. I have simplified them.